jmix-search/search/src/main/java/io/jmix/search/index/fileparsing/FileParserResolver.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2024 Haulmont.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.jmix.search.index.fileparsing;
+
+import org.apache.tika.parser.Parser;
+
+import java.util.List;
+
+/**
+ * Is a part of the extendable engine the gives an ability to implement custom file parser resolvers and to support
+ * custom file types or to modify behavior of existing file parser resolvers.
+ */
+public interface FileParserResolver {
+
+    /**
+     * Returns a collection of supported extensions of the supported file type. E.g. ["xlsx", "XLSX", "DOCX", "DOCX"].
+     * @return collection of supported extensions
+     */
+    List<String> getExtension();
```

Comment:
1. `getSupportedExtensions()`
2. Usually, when we have a collection of beans that may or may not be applied, they either have a method that returns `true/false` whether this bean can be used, e.g. `boolean supports(...)` (`ConditionGenerator`, `AuthenticationLocaleResolver`, etc. just try to search for such method). Another option is to have a `@Nullable` method and the first bean which return a value is used, e.g. `ComponentGenerationStrategy`.

The point is that the calling code should not check _applicability_ itself. And one of the reasons is that the `supports` method can be implemented differently for each bean. 

In this case, the option with the `supports` method is better then `@Nullable` return value. 

Ok, you can leave `getExtension()` method to collect all available extensions for an exception, just rename it to `getSupportedExtensions()`

Reply:
The more flexible engine was implemented. This method was moved to the abstract class. The more general method "getCriteriaDescription()" was described instead of it.